Optimise drawing vu meters

maks opened the pull request:
This improves performance of vumeter drawing up to a point as without vsync we can't really prevent tearing.
It also fixes vumeters to updated when the sequencer is not running for example with midi input or possible other reasons that may be introduced in the future.

This also add "inertia" to the vumeters, giving a pleasing "slower fall off" when volume levels drop suddenly (eg. when playback stops).

Fixes: #454

democloid commented:
First things first. This PR causes my device not to boot.

maks commented:
> First things first. This PR causes my device not to boot.

Strange!! It runs fine on my dev unit. Is it debug or release build? If debug build with optimisation on or off?

democloid commented:
> > First things first. This PR causes my device not to boot.
> 
> Strange!! It runs fine on my dev unit. Is it debug or release build? If debug build with optimisation on or off?

Have you tried after latest changes? Debug unoptimized was my test. Tried multiple times, restarted debugger, etc. Going back to master works fine. No idea!

maks commented:
> > > First things first. This PR causes my device not to boot.
> > 
> > 
> > Strange!! It runs fine on my dev unit. Is it debug or release build? If debug build with optimisation on or off?
> 
> Have you tried after latest changes? Debug unoptimized was my test. Tried multiple times, restarted debugger, etc. Going back to master works fine. No idea!

ah sorry @democloid ! I quickly found what happened, I changed the animation tick rate last minute and didnt notice that I hadn't recompiled to test with it. Fixed now and also same time cleaned up code a bit for it too.
